cluster-autoscaler/clusterstate/utils/node_instances_cache.go

```diff
@@ -92,8 +92,25 @@ func (cache *CloudProviderNodeInstancesCache) removeEntriesForNonExistingNodeGro
 
 // GetCloudProviderNodeInstances returns cloud provider node instances for all node groups returned by cloud provider.
 func (cache *CloudProviderNodeInstancesCache) GetCloudProviderNodeInstances() (map[string][]cloudprovider.Instance, error) {
+	nodeGroups := cache.cloudProvider.NodeGroups()
+
+	// Fetch missing node instances.
+	var wg sync.WaitGroup
+	for _, nodeGroup := range nodeGroups {
+		nodeGroup := nodeGroup
+		if _, found := cache.getCacheEntryLocked(nodeGroup); !found {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				cache.fetchCloudProviderNodeInstancesForNodeGroup(nodeGroup)
```

Comment:
Please add error logging

Reply:
Done

Reply:
The line you added for this always logs even when there is no error
```
node_instances_cache.go:106] Failed to fetch cloud provider node instances for default-worker, error <nil>
```
@krzysztof-jastrzebski @losipiuk

Reply:
Fix in https://github.com/kubernetes/autoscaler/pull/2137